tests/kafkatest/tests/test_upgrade.py

```diff
@@ -0,0 +1,122 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.tests.test import Test
+from ducktape.utils.util import wait_until
+
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.services.kafka import KafkaService, KafkaVersion
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.console_consumer import ConsoleConsumer, is_int
+from kafkatest.services.kafka import kafka_prop
+
+
+class UpgradeTest(Test):
+
+    def __init__(self, test_context):
+        super(UpgradeTest, self).__init__(test_context=test_context)
+
+        self.topic = "test_topic"
+        self.zk = ZookeeperService(test_context, num_nodes=1)
+        self.kafka = KafkaService(test_context, num_nodes=3, zk=self.zk, version=KafkaVersion.V_0_8_2_1, topics={self.topic: {
+                                                                    "partitions": 3,
+                                                                    "replication-factor": 3,
+                                                                    "min.insync.replicas": 2}
+                                                                })
+        self.producer_throughput = 1000
+        self.num_producers = 1
+        self.num_consumers = 1
+
+    def setUp(self):
+        self.zk.start()
+        self.kafka.start()
+
+    def test_upgrade(self):
+        """Test upgrade of Kafka broker cluster from 0.8.2 to 0.9.0
```

Comment:
This is an interesting issue. If we set things up using `KafkaVersion.TRUNK` right now, when 0.9.0 is released, we're going to have to go through and remember to fix up all these tests. Maybe we should proactively add a V_0_9_0_0 and use that instead of TRUNK so it's future proof? We'll still need to change what it points at (i.e. we might have `V_0_0_9_0 = "trunk"` for now, but then switch it at release to `V_0_0_9_0 = "0.9.0.0"` and add a new version to use with `"trunk"` for the next release), but it isolates it to a single location where we need to change version info.